core/templates/pages/skill-editor-page/skill-editor-page.component.ts

```diff
@@ -96,6 +104,78 @@ angular.module('oppia').component('skillEditorPage', {
         return ctrl.skill ? ctrl.skill.getValidationIssues().length : 0;
       };
 
+      ctrl.onClosingSkillEditorBrowserTab = function() {
+        const skill = SkillEditorStateService.getSkill();
+
+        const skillEditorBrowserTabsInfo: EntityEditorBrowserTabsInfo = (
+          LocalStorageService.getEntityEditorBrowserTabsInfo(
+            EntityEditorBrowserTabsInfoDomainConstants
+              .OPENED_SKILL_EDITOR_BROWSER_TABS, skill.getId()));
+
+        if (skillEditorBrowserTabsInfo.doesSomeTabHaveUnsavedChanges() &&
+            UndoRedoService.getChangeCount() > 0) {
+          skillEditorBrowserTabsInfo.setSomeTabHasUnsavedChanges(false);
+        }
+        skillEditorBrowserTabsInfo.decrementNumberOfOpenedTabs();
+
+        LocalStorageService.updateEntityEditorBrowserTabsInfo(
+          skillEditorBrowserTabsInfo,
+          EntityEditorBrowserTabsInfoDomainConstants
+            .OPENED_SKILL_EDITOR_BROWSER_TABS);
+      };
+
+      let createSkillEditorBrowserTabsInfo = function() {
+        const skill = SkillEditorStateService.getSkill();
+
+        let skillEditorBrowserTabsInfo: EntityEditorBrowserTabsInfo = (
+          LocalStorageService.getEntityEditorBrowserTabsInfo(
+            EntityEditorBrowserTabsInfoDomainConstants
+              .OPENED_SKILL_EDITOR_BROWSER_TABS, skill.getId()));
+
+        if (skillEditorBrowserTabsInfo) {
+          skillEditorBrowserTabsInfo.setLatestVersion(skill.getVersion());
+          skillEditorBrowserTabsInfo.incrementNumberOfOpenedTabs();
+        } else {
+          skillEditorBrowserTabsInfo = EntityEditorBrowserTabsInfo.create(
+            'skill', skill.getId(), skill.getVersion(), 1, false);
+        }
+
+        LocalStorageService.updateEntityEditorBrowserTabsInfo(
+          skillEditorBrowserTabsInfo,
+          EntityEditorBrowserTabsInfoDomainConstants
+            .OPENED_SKILL_EDITOR_BROWSER_TABS);
+      };
+
+      let updateSkillEditorBrowserTabsInfo = function() {
+        const skill = SkillEditorStateService.getSkill();
+
+        const skillEditorBrowserTabsInfo: EntityEditorBrowserTabsInfo = (
+          LocalStorageService.getEntityEditorBrowserTabsInfo(
+            EntityEditorBrowserTabsInfoDomainConstants
+              .OPENED_SKILL_EDITOR_BROWSER_TABS, skill.getId()));
+
+        skillEditorBrowserTabsInfo.setLatestVersion(skill.getVersion());
+        skillEditorBrowserTabsInfo.setSomeTabHasUnsavedChanges(false);
+
+        LocalStorageService.updateEntityEditorBrowserTabsInfo(
+          skillEditorBrowserTabsInfo,
+          EntityEditorBrowserTabsInfoDomainConstants
+            .OPENED_SKILL_EDITOR_BROWSER_TABS);
+      };
```

Comment:
Is there a way to combine these two to handle the "initialized" and "reinitialized" scenarios e.g. can it be a createOrUpdateSkillEditorBrowserTabsInfo method? It seems like they do similar things logically. That way we can just trigger "onSkillIntialized" event when the skill is initialized or reinitialized and the method gets called correctly.

Reply:
Done.